apps/server/src/infra/sync/service/sync.service.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock } from '@golevelup/ts-jest';
+import { faker } from '@faker-js/faker';
+import { SyncService } from './sync.service';
+import { TspSyncStrategy } from '../tsp/tsp-sync.strategy';
+import { SyncStrategyTarget } from '../sync-strategy.types';
+
+describe(SyncService.name, () => {
+	let module: TestingModule;
+	let service: SyncService;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				SyncService,
+				{
+					provide: TspSyncStrategy,
+					useValue: createMock<TspSyncStrategy>({
+						getType(): SyncStrategyTarget {
+							return SyncStrategyTarget.TSP;
+						},
+						sync(): Promise<void> {
+							return Promise.resolve();
+						},
+					}),
+				},
+			],
+		}).compile();
+
+		service = module.get(SyncService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('when sync service is initialized', () => {
+		it('should be defined', () => {
+			expect(service).toBeDefined();
+		});
+	});
+
+	describe('startSync', () => {
+		const setup = () => {
```

Comment:
The setup should be on the same level as the `it()` block

Reply:
done ✅